example/src/androidTest/java/com/attentive/example/CreativeUITest.java

```diff
@@ -74,69 +71,91 @@ public static void classTeardown() {
-
-        checkIfCreativeClosed();
+        synchronized (device) {
+            loadCreative(AttentiveConfig.Mode.PRODUCTION);
+
+            UiObject emailInput = device.findObject(selector.resourceId("input0input"));
+            emailInput.setText("testmail@attentivemobile.com");
+
+            // Close the Creative
+            UiObject closeButton = device.findObject(new UiSelector().resourceId("closeIconContaisdfner"));
+            UiObject dismissButton = device.findObject(new UiSelector().descriptionContains("Disdfsmiss this popup"));
+            UiObject clickOutside = device.findObject(new UiSelector().textContains("Attentive Example"));
+            if (closeButton.exists()) {
+                closeButton.click();
+            } else if (dismissButton.exists()) {
+                dismissButton.click();
+            } else {
+                clickOutside.click();
+            }
+
+            checkIfCreativeClosed();
+        }
     }
 
 
     @Test
-    public void loadCreative_fillOutFormAndSubmit_launchesSmsAppWithPrePopulatedText() throws UiObjectNotFoundException {
-        loadCreative(AttentiveConfig.Mode.PRODUCTION);
-
-        // fill in email
-        UiObject emailInput = device.findObject(selector.resourceId("input0input"));
-        emailInput.setText("testmail@attentivemobile.com");
-        device.pressEnter();
-        device.waitForIdle();
-
-        // submit email
-        UiObject emailSubmitButton = device.findObject(selector.resourceId("ctabutton1"));
-        emailSubmitButton.click();
-        device.waitForIdle();
-
-        // click subscribe button
-        UiObject subscribeButton = device.findObject(selector.textContains("GET 10% OFF NOW"));
-        subscribeButton.click();
-        device.waitForIdle();
-
-        // Verify intent to open sms app
-        Intents.intended(allOf(hasAction(ACTION_VIEW), hasData(hasToString(startsWith("sms://")))));
-
-        // Verify that the SMS app is opened with prepopulated text if running locally
-        // (AWS Device Farm doesn't allow use of SMS apps)
-        String testHost = InstrumentationRegistry.getArguments().getString("testHost");
-        if (testHost != null && testHost.equals("local")) {
-            assertTrue(device.wait(Until.hasObject(textContains(SMS_STRING)), 3000));
+    public void loadCreative_fillOutFormAndSubmit_launchesSmsAppWithPrePopulatedText() throws UiObjectNotFoundException, InterruptedException {
+        synchronized (device) {
+            loadCreative(AttentiveConfig.Mode.PRODUCTION);
+
+            // fill in email
+            UiObject emailInput = device.findObject(selector.resourceId("input0input"));
+            emailInput.setText("testmail@attentivemobile.com");
+            device.pressEnter();
+            device.waitForIdle();
+
+            // submit email
+            UiObject emailSubmitButton = device.findObject(new UiSelector().textContains("Continue"));
+            if (emailSubmitButton.exists()) {
```

Comment:
some devices will submit the email on enter, some will not. This is to make sure we proceed those those that don't